flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSource.java

```diff
@@ -111,24 +112,17 @@
 		return kafkaSource;
 	}
 
-	@Override
-	public int getNumberOfFields() {
-		return fieldNames.length;
-	}
-
-	@Override
 	public String[] getFieldsNames() {
-		return fieldNames;
+		return TableSource$class.getFieldsNames(this);
```

Comment:
We do not override this method. As far as I understand we cannot inherit a method from a Scala trait if this trait has implementation: http://stackoverflow.com/a/7637888

Am I missing something?